### PR TITLE
[JENKINS-49322] Add an optional prefix parameter to artifact archiver

### PR DESCRIPTION
**Pull request by segalaj**
Add targetDirectory parameter for the artifact archiver
No test added, I can add some if needed.

See [JENKINS-49322](https://issues.jenkins-ci.org/browse/JENKINS-49322).

### Proposed changelog entries

* Add an optional parent directory parameter to the artifact archiver

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

no idea, sorry 😉

### CODE REVIEW

**Comment by daniel-beck** [comment]
Alternative to #1878, as JENKINS-49322 duplicates JENKINS-12379.

File: core/src/main/java/jenkins/model/ArtifactManager.java [hunk 1]
```diff
@@ -60,11 +60,12 @@
      * @param launcher a launcher to use if external processes need to be forked
      * @param listener a way to print messages about progress or problems
      * @param artifacts map from paths in the archive area to paths relative to {@code workspace} (all paths {@code /}-separated)
+     * @param targetDirectory the target parent directory
      * @throws IOException if transfer or copying failed in any way
      * @throws InterruptedException if transfer was interrupted
      * @see ArtifactArchiver#perform(Run, FilePath, Launcher, TaskListener)
      */
-    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts) throws IOException, InterruptedException;
+    public abstract void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String,String> artifacts, String targetDirectory) throws IOException, InterruptedException;
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/ArtifactManager.java]: Change breaking API compatibility.

**Review by daniel-beck — CHANGES_REQUESTED** [review]

File: core/src/main/java/jenkins/model/ArtifactManager.java (hunk 1 above)
Comment (segalaj) [reply core/src/main/java/jenkins/model/ArtifactManager.java]: @daniel-beck I agree. Should I add another method signature or find antoher way that avoid modifying the API?

**Comment by batmat** [comment]
This is conflicted, has a change request from Daniel, and last comment was 1+ year ago.

@segalaj do you plan to get back to this? No worries if not, this happens :). Just trying to push open PRs to completion.

I will propose this for closing and will close it in the next days if the status doesn't change.

Thanks!

**Comment by segalaj** [comment]
@batmat thanks for your comment, not sure I'll come back on this. I was expecting an answer to my last comment to continue.

**Comment by daniel-beck** [comment]
> I was expecting an answer to my last comment to continue.

Sorry about that, I missed your response.

File: core/src/main/java/jenkins/model/ArtifactManager.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/jenkins/model/ArtifactManager.java]: Yes. Add another method with more arguments, and invoke that one from this one with whatever is the "default" argument.

**Comment by batmat** [comment]
@segalaj Daniel was able to answer in https://github.com/jenkinsci/jenkins/pull/3284#discussion_r272168524. Sorry again for the miss.

Do you think you want to keep this open, so you can finalize it, or you'd prefer to close it even temporarily (and we can reopen anytime you want to get back to it)?

Thank you very much and sorry again.

**Comment by batmat** [comment]
Given this PR is conflicted, and I didn't get answer from my last comment 11 days, I am going to close it.

Please feel free to comment so we reopen this PR anytime once/if you are available again. Or filing a new PR depending on your preference.

Thanks a lot.

**Pull request closed without merging**